src/source-af-packet.c

```diff
@@ -1541,6 +1541,8 @@ TmEcode ReceiveAFPLoop(ThreadVars *tv, void *data, void *slot)
          * us from alloc'ing packets at line rate */
         PacketPoolWait();
 
+        // Indicate that the thread is actually running its application level code (i.e., it can poll packets)
+        TmThreadsSetFlag(tv, THV_RUNNING);
```

Comment:
`TmThreadSetFlag` wraps around an atomic operation, so it is relatively expensive. Running that from this loop seems excessive.

Reply:
Yeah; I was trying to get them as close as possible to the first poll for packets.  I'll pull them out to before the loop.